Formula/openttd.rb

```diff
@@ -39,18 +52,31 @@ class Openttd < Formula
   end
 
   def install
-    mkdir "build" do
-      system "cmake", "..", *std_cmake_args
-      system "make"
-      system "cpack || :"
+    args = std_cmake_args
+    args << "-DCMAKE_INSTALL_BINDIR=bin" unless OS.mac?
+    system "cmake", "-G", "Ninja", "-S", ".", "-B", "build", *args
+    system "ninja", "-C", "build"
+    if OS.mac?
+      cd "build" do
+        system "cpack || :"
+      end
+    else
+      system "ninja", "-C", "build", "install"
     end
 
     app = "build/_CPack_Packages/amd64/Bundle/openttd-#{version}-macos-amd64/OpenTTD.app"
     resources.each do |r|
-      (buildpath/"#{app}/Contents/Resources/baseset/#{r.name}").install r
+      if OS.mac?
+        (buildpath/"#{app}/Contents/Resources/baseset/#{r.name}").install r
+      else
+        (share/"baseset"/r.name).install r
```

Comment:
Were you able to start the game locally on Linux? [The game looks in certain paths for files](https://github.com/OpenTTD/OpenTTD/blob/12.2/docs/directory_structure.md#openttd-directory-structure). One of them is installation directory on Linux, which should match CMake `GLOBAL_DIR` variable equal to [`${CMAKE_INSTALL_FULL_DATADIR}/${BINARY_NAME}`](https://github.com/OpenTTD/OpenTTD/blob/12.2/cmake/Options.cmake#L19)

The [`CMAKE_INSTALL_DATADIR `](https://github.com/OpenTTD/OpenTTD/blob/12.2/CMakeLists.txt#L30) defaults to `share/games` and the full variant is just the absolute path `#{share}/games`. So, the game should automatically search `#{share}/games/openttd` for a `baseset` directory.

This was why I had used `share/"games/openttd/baseset"/r.name` instead, but I only tested on macOS.

Another test that can be run without actually starting game is to use `openttd -h` and make sure the files show up:
```
List of graphics sets:
           OpenGFX: OpenGFX base graphics set for OpenTTD. Freely available under the terms of the GNU General Public License version 2. [OpenGFX 7.1]
      original_dos: Original Transport Tycoon Deluxe DOS edition graphics. (unusable: 5 missing files)
  original_windows: Original Transport Tycoon Deluxe Windows edition graphics. (unusable: 5 missing files)

List of sounds sets:
      original_dos: Original Transport Tycoon Deluxe DOS edition sounds. (unusable: 1 missing file)
           NoSound: A sound pack without any sounds.
           OpenSFX: OpenSFX base sound set for OpenTTD. Freely available under the terms of these licences: CC BY-SA 3.0, GPLv2 (or later) and CDDL 1.1. For full credits see "readme.txt". [OpenSFX 1.0.3]

List of music sets:
      original_tto: Original Transport Tycoon (Original/World Editor) DOS edition music. (unusable: 17 missing files)
           OpenMSX: OpenMSX base music set for OpenTTD. Freely available under the terms of the GPL v2. For full credits see "readme.txt" [OpenMSX 0.4.2]
      original_dos: Original Transport Tycoon Deluxe DOS edition music. (unusable: 22 missing files)
           NoMusic: A music pack without actual music.
  original_windows: Original Transport Tycoon Deluxe Windows edition music. (unusable: 22 missing files)
```

Reply:
To match what is done in Arch Linux: https://github.com/archlinux/svntogit-community/blob/packages/openttd/trunk/PKGBUILD, I set `CMAKE_INSTALL_DATADIR` to be `#{share}` so that we can install the resources to `#{share}/openttd/baseset`.  I've confirmed that it can find the files - I get the same output on Linux you posted above.